File: Assignment 2/Q9.py

n = int(input("Enter an integer: "))
on = n
rn = 0
while n != 0:
    r = n % 10
    rn = rn * 10 + r
    n //= 10

if on == rn:
    print(on,"is a palindrome.")
else:
    print(on,"is not a palindrome.")

File: Assignment 2/Q9.c
#include <stdio.h>

void main() {
    int n, rn = 0, r, on;

    printf("Enter an integer: ");
    scanf("%d", &n);

    on = n;

    while (n != 0) {
        r = n % 10;
        rn = rn * 10 + r;
        n /= 10;
    }

    if (on == rn) {
        printf("%d is a palindrome.\n", on);
    } else {
        printf("%d is not a palindrome.\n", on);
    }
    }
